mobsf/MobSF/views/authorization.py

```diff
@@ -0,0 +1,169 @@
+"""User management and authorization."""
+from itertools import chain
+from inspect import signature
+from functools import wraps
+
+from django.contrib.auth.models import User
+from django.contrib.auth.models import Group, Permission
+from django.shortcuts import (
+    redirect,
+    render,
+)
+from django.contrib import messages
+from django.contrib.auth import get_user_model
+from django.contrib.admin.views.decorators import staff_member_required
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.decorators import permission_required as pr
+from django.views.decorators.http import require_http_methods
+from django.template.defaulttags import register
+from django.conf import settings
+
+from mobsf.MobSF.forms import RegisterForm
+from mobsf.MobSF.utils import (
+    USERNAME_REGEX,
+    get_md5,
+)
+from mobsf.DynamicAnalyzer.views.common.shared import (
+    send_response,
+)
+
+register.filter('md5', get_md5)
+PERM_CAN_SCAN = 'can_scan'
+PERM_CAN_SUPPRESS = 'can_suppress'
+PERM_CAN_DELETE = 'can_delete'
+PERMISSIONS = {
+    'SCAN': f'StaticAnalyzer.{PERM_CAN_SCAN}',
+    'SUPPRESS': f'StaticAnalyzer.{PERM_CAN_SUPPRESS}',
+    'DELETE': f'StaticAnalyzer.{PERM_CAN_DELETE}',
+}
+DJANGO_PERMISSIONS = {
+    'SCAN': (PERM_CAN_SCAN, 'Scan Files'),
+    'SUPPRESS': (PERM_CAN_SUPPRESS, 'Suppress Findings'),
+    'DELETE': (PERM_CAN_DELETE, 'Delete Scans'),
+}
+MAINTAINER_GROUP = 'Maintainer'
+VIEWER_GROUP = 'Viewer'
+
+
+def permission_required(perm):
+    def decorator(view):
+        @wraps(view)
+        def wrapper(request, *args, **kwargs):
+            sig = signature(view)
+            arguments = sig.bind(request, *args, **kwargs)
+            api = arguments.arguments.get('api')
+            if settings.DISABLE_AUTHENTICATION == '1' or api:
+                # Disable authorization for all functions
+                return view(request, *args, **kwargs)
+            # Enforce authorization for all web function calls
+            return pr(
+                perm,
+                raise_exception=True)(view)(request, *args, **kwargs)
```

Comment:
This should be 403 Forbidden, but since after changing django `DEBUG=True` It's returning `500` for exceptions. I will debug and check what the actual exception is.

Reply:
This one looks like an exception since the user doesn't have admin roles and we are not enabling the django admin app. I made a template to handle 500 errors now.